src/TrackJS/TrackJSEmbed.php

```diff
@@ -74,7 +74,7 @@ public function getEmbedHtml () : string
         $trackJsToken = $this->hostingConfig->getTrackJsToken();
 
         // only embed if token is set, in production and not in debug
-        if (null === $trackJsToken || $this->isDebug || "prod" !== $this->environment)
+        if (null === $trackJsToken || $this->isDebug || !\in_array($this->hostingConfig->getDeploymentTier(), ["production", "staging"]))
```

Comment:
```suggestion
        if (null === $trackJsToken || $this->isDebug || !\in_array($this->hostingConfig->getDeploymentTier(), ["production", "staging"], true))
```

But thinking about it.. We should 

a) validate the deployment tier to either `development`, `staging`, `production` (so no other values are allowed)
b) add a convenience getter for exactly this check here.